src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerConfig.java

```diff
@@ -69,4 +70,22 @@
     super(CONFIG, props, false);
   }
 
+  /**
+   * This method won't call {@link Configurable#configure(Map)} for {@link LiKafkaProducerConfig#AUDITOR_CLASS_CONFIG}
+   * during instance creation
+   */
+  @Override
+  public <T> T getConfiguredInstance(String key, Class<T> t) {
+    if (AUDITOR_CLASS_CONFIG.equals(key)) {
```

Comment:
@apakulov Thanks for the quick response. Yes, I think the first one is cleaner. We don't need to to have this in `LiKafkaConsumerConfig`.